crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -1261,7 +1260,8 @@ impl<'src> Parser<'src> {
             });
         }
 
-        let first_element = self.parse_named_expression_or_higher();
+        // Parse the first element with a more general rule and limit it later.
+        let first_element = self.parse_star_expression_or_higher(AllowNamedExpression::Yes);
```

Comment:
What we will be doing is to use a more general rule for the first element as we don't yet know whether it's a comprehension or a normal list. For comprehension, we'll report an error later if a starred expression is being used.

Reply:
You say that we report an error later. I wasn't able to spot the error reporting right away. Is this something you plan to add later (including tests)?

Reply:
Yes, it was added in the list comprehension PR. Sorry, I should've made that clear.

Reply:
Linking for posterity: https://github.com/astral-sh/ruff/pull/10634/files#diff-b0faf8ec2615ebd31aff987f04757ead6fcfef23e995cd370e5da1700e9ae684